src/hip/hip_build_utils.cpp

```diff
@@ -139,25 +145,28 @@ boost::filesystem::path HipBuild(boost::optional<TmpDir>& tmp_dir,
 
         return hsaco->path();
     }
-#elif OFFLOADBUNDLER_BIN
-    if(IsClangXXCompiler())
+    else
+#endif
+#ifdef OFFLOADBUNDLER_BIN
+        if(IsClangXXCompiler())
     {
         // call clang-offload-bundler
         tmp_dir->Execute(OFFLOADBUNDLER_BIN,
-                         "-type=o -targets=hip-amdgcn-amd-amdhsa-" + dev_name + " -inputs=" +
-                             bin_file.string() + " -unbundle");
+                         "--type=o --targets=hip-amdgcn-amd-amdhsa-" + dev_name + " --inputs=" +
+                             bin_file.string() + " --outputs=" + bin_file.string() +
+                             ".hipfb --unbundle");
```

Comment:
sorry my previous comments caused some confusion. Here the input file is a fatbin, the output file is a code object. So let's better name the output as .hsaco. thanks.

Reply:
done